powerdnsadmin/routes/index.py

```diff
@@ -1056,8 +1056,12 @@ def create_group_to_account_mapping():
 
 
 def handle_account(account_name, account_description=""):
+    if Setting().get('account_name_extra_chars'):
```

Comment:
Could you please place the account name sanitization logic into a dedicated function which you call from within both `models.account.Account.__init__` and `routes.index.handle_account`? E.g. roughly:

```python3
class Account(db.Model)
    # ...
    def __init__(self, name=None, description=None, contact=None, mail=None):
        self.name = Account.sanitize_name(name) if name is not None else name
        # ...
    # ...

    
    @staticmethod
    def sanitize_name(name):
        allowed_characters = "abcdefghijklmnopqrstuvwxyz0123456789"
        if Setting().get('account_name_extra_chars'):
            allowed_characters += "_-."
        sanitized_name = ''.join(c for c in name.lower() if c in allowed_characters)
        if not sanitized_name:
            raise InvalidAccountNameException("Empty string is not a valid account name")
        return sanitized_name
```
That should make intent clearer and also unifies the logic into one sensible place, improving maintainability.

Reply:
The amazing thing with you is that you provide the code you asked to change !